src/blake2-impl.h

```diff
@@ -141,6 +141,9 @@ static inline void secure_zero_memory(void *v, size_t n)
 {
 #if defined(_WIN32) || defined(WIN32)
   SecureZeroMemory(v, n);
+#elif defined(__hpux)
+  static void *(*const volatile memset_v)(void *, int, size_t) = &memset;
+  memset_v(v, 0, n);
```

Comment:
Why not just `memset(v, 0, n)`?

I suppose it is to avoid using compiler builtin memset.
But I would like to explicit comment when hack like this is needed.

Reply:
As far as I understand the code it guarantees not to be optimized away because the pointer `v` is not used after the `memset()`. The previous behavior is portable that's why I have reintroduced it.